enus number wiring for the 6.1 rules (Mudb0y money/hyphen)

Retires IBM's money rules under _43 names (kept for DM marks, FF
francs and the other currencies 6.1 no longer knows); u_integ tries
6.1's rule first, the _43 on decline, and money_range after either;
u_vars initialises the currency-kind and hyphen-chain fields;
et_norm hooks hyphen_in_chain into eng_hyphen_between_nums.

diff --git a/lang/enus/rules/et_norm.dr b/lang/enus/rules/et_norm.dr
index 376e31e..2996d50 100644
--- a/lang/enus/rules/et_norm.dr
+++ b/lang/enus/rules/et_norm.dr
@@ -4148,6 +4148,15 @@ label L0 was _eng_hyphen_between_nums
   push state 0
   call ZZstarttest1 arity 1 depth 16
   popn 16
+  load movl slotaddr -68 into r0
+  push slotaddr -68
+  load movl slotaddr -60 into r0
+  push slotaddr -60
+  push state 0
+  call hyphen_in_chain arity 3 depth 3
+  popn 3
+  cmp testl reg r0 reg r0
+  branch je to L7
   load movl slotaddr -60 into r0
   push slotaddr -60
   load movl slotaddr -28 into r0
diff --git a/lang/enus/rules/et_numbr.dr b/lang/enus/rules/et_numbr.dr
index be592ff..541b9ae 100644
--- a/lang/enus/rules/et_numbr.dr
+++ b/lang/enus/rules/et_numbr.dr
@@ -1,7 +1,7 @@
 # The rules of et_numbr.obj, written by tools/rules/notation.py.
 # One operation to a line. See docs/rules.md.
 
-rule add_eng_currency_units from et_numbr.obj
+rule add_eng_currency_units_43 from et_numbr.obj
 shape frame 208 argbase 8 params 3
 label L0 was _add_eng_currency_units
   alu andl imm 0 slot -16
@@ -233,7 +233,7 @@ label L21 was at_0x293
   return reg r0
 end
 
-rule add_eng_subunits from et_numbr.obj
+rule add_eng_subunits_43 from et_numbr.obj
 shape frame 208 argbase 8 params 3
 label L0 was _add_eng_subunits
   alu andl imm 0 slot -8
@@ -4381,7 +4381,7 @@ label L39 was at_0x477
 label L40 was $L105313
 end
 
-rule add_eng_point_digits from et_numbr.obj
+rule add_eng_point_digits_43 from et_numbr.obj
 shape frame 304 argbase 116 params 3
 label L0 was _add_eng_point_digits
   load movl imm 0 into r7
@@ -11924,7 +11924,7 @@ label L22 was at_0x2f2
 label L23 was $L106039
 end
 
-rule convert_eng_monetary_exp from et_numbr.obj
+rule convert_eng_monetary_exp_43 from et_numbr.obj
 shape frame 300 argbase 108 params 5
 label L0 was _convert_eng_monetary_exp
   load movl imm 0 into r3
@@ -12841,7 +12841,7 @@ label L78 was $X75$100205
   load movl slotaddr 84 into r0
   push slotaddr 84
   push reg r6
-  call add_eng_currency_units arity 3 depth 4
+  call add_eng_currency_units_43 arity 3 depth 4
   popn 4
   jump to L82
 label L79 was $X76$100215
@@ -12874,7 +12874,7 @@ label L80 was $X77$100221
   load movl slotaddr 92 into r0
   push slotaddr 92
   push reg r6
-  call add_eng_currency_units arity 3 depth 4
+  call add_eng_currency_units_43 arity 3 depth 4
   popn 4
 label L81 was $X78$100224
   load movl ind reg r6 2200 into r0
@@ -13021,7 +13021,7 @@ label L90 was $X38$100058
   load movl slotaddr 84 into r0
   push slotaddr 84
   push reg r6
-  call add_eng_point_digits arity 3 depth 4
+  call add_eng_point_digits_43 arity 3 depth 4
   popn 4
 label L91 was $X39$100064
   push reg r6
@@ -13110,7 +13110,7 @@ label L94 was $X41$100076
   load movl slotaddr 48 into r0
   push slotaddr 48
   push reg r6
-  call add_eng_currency_units arity 3 depth 4
+  call add_eng_currency_units_43 arity 3 depth 4
   popn 4
 label L95 was $X43$100094
   load movl slot 52 into r0
@@ -13134,7 +13134,7 @@ label L97 was $X84$100239
   load movl slotaddr 92 into r0
   push slotaddr 92
   push reg r6
-  call add_eng_subunits arity 3 depth 4
+  call add_eng_subunits_43 arity 3 depth 4
   popn 4
 label L98 was $X86$100256
   push reg r6
@@ -14197,7 +14197,7 @@ label L21 was at_0x280
 label L22 was $L106257
 end
 
-rule eng_monetary_exp from et_numbr.obj
+rule eng_monetary_exp_43 from et_numbr.obj
 shape frame 232 argbase 116 params 3
 label L0 was _eng_monetary_exp
   load movl imm 0 into r3
@@ -14366,7 +14366,7 @@ label L8 was at_0x19a
   load movl slotaddr 96 into r0
   push slotaddr 96
   push reg r6
-  call convert_eng_monetary_exp arity 5 depth 10
+  call convert_eng_monetary_exp_43 arity 5 depth 10
   popn 10
   cmp testl reg r0 reg r0
   branch jne to L2
diff --git a/lang/enus/rules/u_integ.dr b/lang/enus/rules/u_integ.dr
index 8762b74..deafeb0 100644
--- a/lang/enus/rules/u_integ.dr
+++ b/lang/enus/rules/u_integ.dr
@@ -610,7 +610,22 @@ label L0 was _monetary_exp
   call eng_monetary_exp arity 3 depth 5
   popn 5
   cmp testl reg r0 reg r0
+  branch je to L5
+  load movl slotaddr -20 into r0
+  push slotaddr -20
+  load movl slotaddr -12 into r0
+  push slotaddr -12
+  push state 0
+  call eng_monetary_exp_43 arity 3 depth 5
+  popn 5
+  cmp testl reg r0 reg r0
   branch jne to L1
+label L5
+  load movl slotaddr -20 into r0
+  push slotaddr -20
+  push state 0
+  call money_range arity 2 depth 2
+  popn 2
   store movl imm 1 slot -4
 label L1 was at_0xa9
   load movl slotaddr -4 into r0
@@ -1182,6 +1197,11 @@ label L0 was _convert_monetary_exp
   popn 7
   cmp testl reg r0 reg r0
   branch jne to L1
+  load movl slotaddr -20 into r0
+  push slotaddr -20
+  push state 0
+  call money_range arity 2 depth 2
+  popn 2
   store movl imm 1 slot -4
 label L1 was at_0xcd
   load movl slotaddr -4 into r0
diff --git a/lang/enus/rules/u_vars.dr b/lang/enus/rules/u_vars.dr
index 91f8590..6beadaa 100644
--- a/lang/enus/rules/u_vars.dr
+++ b/lang/enus/rules/u_vars.dr
@@ -1541,6 +1541,7 @@ label L2 was at_0x59
   store movw reg r7w ind reg r6 1586
   store movw reg r7w ind reg r6 1590
   store movw reg r7w ind reg r6 342
+  store movl reg r7 ind reg r6 4260
   call ZZstarttest1 arity 1 depth 2
   push reg r6
   call ZZlpta_load_vvg__setscan_0001r__1 arity 1 depth 3
@@ -2131,6 +2132,13 @@ label L3 was at_0x2d9
   store movw imm 9 ind reg r6 1910
   store movw imm 10 ind reg r6 1914
   store movw imm 11 ind reg r6 1918
+  store movw imm 13 ind reg r6 4266
+  store movw imm 1 ind reg r6 4270
+  store movw imm 2 ind reg r6 4274
+  store movw imm 3 ind reg r6 4278
+  store movw imm 4 ind reg r6 4282
+  store movw imm 5 ind reg r6 4286
+  store movw imm 8 ind reg r6 4290
   store movw reg r7w ind reg r6 1210
   store movw imm 2 ind reg r6 1214
   store movw reg r3w ind reg r6 1218
